### PR TITLE
fix: Fix variable being cleared after reboot by save button ID to DB

### PR DESCRIPTION
**Pull request by iach526526**
fix issues : #89 
I create a new table storage detail 

## Summary by CodeRabbit

- **New Features**
  - Introduced database integration for managing button attributes and user rewards.
  - Added environment variable support for enhanced database configuration.

- **Improvements**
  - Enhanced bot's intent settings for improved access to member information and message content.
  - Improved method naming for better clarity.
  - Added a new asynchronous function for logging button data during gift dispatch.

- **Bug Fixes**
  - Resolved issues related to the handling of button interactions and gift tracking.

- **Chores**
  - Updated project dependencies for enhanced functionality and compatibility.
  - Added new packages for environment variable management.

### CODE REVIEW

File: cog/admin_role.py [hunk 1]
```diff
@@ -55,14 +55,22 @@ def __reward(uid, user_name, bouns_type, bouns):
             write(uid, bouns_type, current_point + bouns, cursor)
             end(connection, cursor)
             print(f"{uid} {user_name} get {bouns} {bouns_type} by Gift")
-
+        # 存資料庫存取按鈕屬性(包括獎勵類型、數量)
+        def __readDB(self, btnid):
```
Comment (winstonsung) [inline cog/admin_role.py:59]: (From Pylint)

Didn't meet the naming convension.

Could rename to `__read_db` instead.

File: cog/admin_role.py [hunk 2]
```diff
@@ -55,14 +55,22 @@ def __reward(uid, user_name, bouns_type, bouns):
             write(uid, bouns_type, current_point + bouns, cursor)
             end(connection, cursor)
             print(f"{uid} {user_name} get {bouns} {bouns_type} by Gift")
-
```
Comment (winstonsung) [inline cog/admin_role.py:58]: 1 blank line should be kept before functions/methods.

File: cog/admin_role.py (hunk 1 above)
Comment (winstonsung) [inline cog/admin_role.py:59]: (From CodeRabbit AI)

Missing type annotation `btnid: int`.

Comment (winstonsung) [inline cog/admin_role.py:59]: (Could ignore)

Consider rename parameter `btnid` to `btn_id`.

File: cog/admin_role.py [hunk 3]
```diff
@@ -113,11 +121,16 @@ async def fetch_user_by_name(name):
                 color = discord.Color.blurple()
             )
 
+            async def recordDB(btnid, type, count,recipient):
```
Comment (winstonsung) [inline cog/admin_role.py:124]: (From Pylint)

Didn't meet the naming convension.

Could rename to __record_db instead.

Comment (winstonsung) [inline cog/admin_role.py:124]: (From CodeRabbit AI)

Missing type annotations `(btnid: int, type: str, count: int, recipient: str) -> None:`.

**Review by winstonsung — CHANGES_REQUESTED** [review]

File: cog/admin_role.py [hunk 4]
```diff
@@ -17,72 +14,80 @@ async def on_ready(self) -> None:
     # 成員身分組
     class RoleView(discord.ui.View):
         def __init__(self):
-            super().__init__(timeout = None) # timeout of the view must be set to None
+            super().__init__(timeout=None)  # timeout of the view must be set to None
```
Comment (winstonsung) [inline cog/admin_role.py:17]: Unexpected whitespace changes.

**Review by winstonsung — CHANGES_REQUESTED** [review]

**Comment by iach526526** [comment]
@winstonsung why pylint always says Unable to import 'dotenv' ?

**Comment by winstonsung** [comment]
Because the Pylint GitHub Actions workflow haven't switch to `-r requirements.txt` .

File: cog/admin_role.py (hunk 4 above)
Comment (winstonsung) [reply cog/admin_role.py]: Correction: Coding conventions

File: requirements.txt [hunk 5]
```diff
@@ -34,11 +35,12 @@ pyasn1_modules==0.4.0
...
 pyparsing==3.1.2
+python-dotenv==1.0.1
 requests==2.32.3
 rsa==4.9
 tqdm==4.66.4
 typing_extensions==4.12.2
 uritemplate==4.1.1
-urllib3==2.2.2
-Werkzeug==3.0.3
+urllib3==2.2.1
```
Comment (winstonsung) [inline requirements.txt:44]: Looks like there are some downgrading. Is it intended?

File: requirements.txt [hunk 6]
```diff
@@ -34,11 +35,12 @@ pyasn1_modules==0.4.0
...
+Werkzeug==3.0.1
```
Comment (winstonsung) [inline requirements.txt:45]: Looks like there are some downgrading. Is it intended?

**Comment by winstonsung** [comment]
Is the downgrading from

```
urllib3==2.2.2 
werkzeug==3.0.3 
```

to

```
urllib3==2.2.1 
werkzeug==3.0.1
```

intended?

File: requirements.txt (hunk 6 above)
Comment (winstonsung) [reply requirements.txt]: @iach526526 wrote on SCAICT Discord server:
> `requirements.txt` 我用指令自動生的，是我兩台電腦版本不同才造成這樣的結果
> 我沒有刻意去降這兩個套件的版本

File: .gitignore
Comment (winstonsung) [inline .gitignore:1]: (TODO: Fix missing blank line at the end of the file.)

**Review by winstonsung — APPROVED** [review]

File: .gitignore
Comment (winstonsung) [reply .gitignore:1]: (Fixed)

**Pull request merged**